lnd.go

```diff
@@ -1124,6 +1130,13 @@ func waitForWalletPassword(cfg *Config, restEndpoints []net.Addr,
 	// The wallet has already been created in the past, and is simply being
 	// unlocked. So we'll just return these passphrases.
 	case unlockMsg := <-pwService.UnlockMsgs:
+
```

Comment:
I was still able to reproduce the issue with the current changes. The issue doesn't seem to be related to the init call being asynchronous, but rather with the `UnlockerService` having a queued `UnlockWallet` call that it doesn't cancel after `InitWallet` has been called. If we had a `UnlockerService.Quit` method, we could call it here and check that it's been closed after the lock has been acquired at the `UnlockerService` level.